### PR TITLE
RHTAPINST-39: Embedded Helm Charts

### PR DESCRIPTION
**Pull request by otaviof**
Embedding the Helm Charts and other resources needed for the installation process in the final executable `rhtap-cli`. The flag `--embedded` is enabled by default, and can be set to `false` in order to use local installer resources.

Golang `embed` doesn't allow including symbolic links and files starting with `_` (underscore), thus the approach chosen for this PR is using a tarball from the `installer` directory and navigating through it as a regular `fs.FS` interface.

```bash
bin/rhtap-cli deploy --embedded=true
```

It's also possible to list and extract embedded contents using:

```bash
bin/rhtap-cli installer --list

mkdir /var/tmp/rhtap-cli-resources
bin/rhtap-cli installer --extract /var/tmp/rhtap-cli-resources
```

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -16,6 +16,17 @@ After you have cloned the repository and installed the requirements, you can sta
 make
 ```
 
+The resources needed for the installer are stored in the `installer` directory, you can find the `config.yaml` file there. Before the application is built, the contents of the installer directory are package into a tarball and embedded into the application binary.
```
Comment (Roming22) [inline CONTRIBUTING.md]: *are packaged

File: Dockerfile [hunk 2]
```diff
@@ -8,13 +8,13 @@ USER root
 
 WORKDIR /workdir/rhtap-cli
 
-COPY charts/ ./charts/
```
Comment (Roming22) [inline Dockerfile:11]: Why not also keep the resources in the image so users can easily customize the deployment if necessary?

File: README.md [hunk 3]
```diff
@@ -102,32 +102,34 @@ Please refer to the [CONTRIBUTING.md](CONTRIBUTING.md) file for more information
...
-4.Run the command `rhtap-cli integration` to provide integrations to external components.
-  The below command will list the options supported. 
+4. Run the command `rhtap-cli integration` to provide integrations to external components. The command below lists the options supported: 
   
 ```bash
 rhtap-cli integration --help
 ```
   
-5.Finally run the below command to proceed with RHTAP deployment. 
+5. Finally run the below command to proceed with RHTAP deployment.
```
Comment (Roming22) [inline README.md:132]: Should the example use `--embedded=false --config=config.yaml`, as the user might have modified the config.yaml?

File: pkg/subcmd/deploy.go [hunk 4]
```diff
@@ -36,6 +35,9 @@ applied, on the attribute 'rhtapCLI.dependencies[]'.
 
 The platform configuration is rendered from the values template file
 (--values-template), this configuration payload is given to all Helm charts.
+
+The installer resources are embedded on the executable, these resources are
```
Comment (Roming22) [inline pkg/subcmd/deploy.go]: *embedded in

File: pkg/subcmd/template.go [hunk 5]
```diff
@@ -44,6 +42,9 @@ chart directory, optional.
 
 Additionally, the '--debug' flag should be used to display the raw values template
 payload, regardless of whether it is valid YAML or not.
+
+The installer resources are embedded on the executable, these resources are
```
Comment (Roming22) [inline pkg/subcmd/template.go]: *embedded in

File: Dockerfile (hunk 2 above)
Comment (otaviof) [reply Dockerfile:11]: These resources are on the `installer` directory, copied into the image accordingly.

File: README.md (hunk 3 above)
Comment (otaviof) [reply README.md:132]: The `config.yaml` isn't embedded, to this file is still required. Only charts and scripts are embedded currently.

**Review by Roming22 — APPROVED** [review]

**Comment by Roming22** [comment]
@otaviof You need to update `yamllint.yaml`, and take care of sonarcloud and snyk before it can be merged.

**Comment by Roming22** [comment]
@otaviof PR is still broken.

**Comment by otaviof** [comment]
> @otaviof You need to update `yamllint.yaml`, and take care of sonarcloud and snyk before it can be merged.

The SonarCloud alerts have been triggered after moving files into the `installer` directory, have we ignored these alerts before? Or, perhaps, this is a new static code analysis? Well, anyway, given this PR is not modifying these files, we should have a dedicated PR to fix the issues. We can ignore them for the time being, but fix right after.

**Comment by otaviof** [comment]
> > @otaviof You need to update `yamllint.yaml`, and take care of sonarcloud and snyk before it can be merged.
> 
> The SonarCloud alerts have been triggered after moving files into the `installer` directory, have we ignored these alerts before? Or, perhaps, this is a new static code analysis? Well, anyway, given this PR is not modifying these files, we should have a dedicated PR to fix the issues. We can ignore them for the time being, but fix right after.

@Roming22, I've included on more commit to fix the styling part of SonarCloud alerts, and documented the rest using the web interface. Please consider.

**Comment by flacatus** [comment]
due that we cannot test yet in konflux task changes we test manually changing the task to otavio fork and the test pass!

Screenshot from 2024-09-11 10-27-12

**Comment by otaviof** [comment]
> due that we cannot test yet in konflux task changes we test manually changing the task to otavio fork and the test pass!
> 
> Screenshot from 2024-09-11 10-27-12

I'm going to remove the last commit, with the temporary settings, and then we can merge this afterwards. Thanks!

**Pull request merged**